### DISCUSSION TITLE
How to commit immediately

### DISCUSSION THREAD

**Discussion by hyry1121** [wtertinek/Linq2Acad#15] [Q&A]
example:

        var i = 0;
        using var db = AcadDatabase.Active();
        while( true ) {
          var p3d = getPoint3d( "select position" );
          if( p3d == null ) {
            break;
          }

          var ent = new DBText {
            TextString = $"{i++}",
            Position = p3d
          };
          db.ModelSpace.Add( ent );
        }

the example will be committed all after the loop break.
my question is: How to make a commit in every loop ? it means real-time committing.
Thanks!

**Comment by wtertinek** [comment]
Hi,

you're right, the current implementation does only one commit when disposing of the AcadDatabase object. Currently it's not intended to implement intermediate commits. I think intermediate commits would complicates the API.

One way to solve your problem would be to move the AcadDatabase instanciation into the while loop:

```cs
var i = 0;

while(true)
{
  var p3d = getPoint3d("select position");
  
  if( p3d == null )
  {
    break;
  }

  (using var db = AcadDatabase.Active())
  {
    var ent = new DBText
              {
                TextString = $"{i++}",
                Position = p3d
              };
    db.ModelSpace.Add(ent);
  }
}
```

Hope that helps.

-Wolfgang
